internal/telemetry/client.go

```diff
@@ -131,87 +138,91 @@ type Client struct {
 	// metrics are sent
 	metrics    map[string]*metric
 	newMetrics bool
+
+	// logLock gaurds the logging field
+	logLock sync.RWMutex
+	// Logging allows us to toggle on agentless in the case where
+	// there are issues with sending telemetry to the agent
+	Logging bool
 }
 
 func (c *Client) log(msg string, args ...interface{}) {
-	if c.Logger == nil {
+	// we don't log if the client is temporarily using agentless
+	// to avoid spamming the user with instrumentation telemetry error messages
+	c.logLock.RLock()
+	defer c.logLock.RUnlock()
+	if !c.Logging {
 		return
 	}
-	c.Logger.Printf(msg, args...)
+	log.Warn(fmt.Sprintf(LogPrefix+msg, args...))
+}
+
+// logging is used to turn logging on/off
+func (c *Client) logging(logging bool) {
+	c.logLock.Lock()
+	defer c.logLock.Unlock()
+	c.Logging = logging
+}
+
+// Started returns whether the client has started or not
+func (c *Client) Started() bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.started
 }
 
 // Start registers that the app has begun running with the given integrations
 // and configuration.
-func (c *Client) Start(integrations []Integration, configuration []Configuration) {
+// TODO: implement passing in error information about tracer start
+func (c *Client) Start(configuration []Configuration) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if c.Disabled || !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", true) {
+	if !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", true) || c.Disabled {
 		return
 	}
 	if c.started {
 		return
 	}
-	c.debug = internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_DEBUG", c.debug)
-
 	c.started = true
 
 	// XXX: Should we let metrics persist between starting and stopping?
 	c.metrics = make(map[string]*metric)
+	c.applyDefaultOps()
```

Comment:
I wonder if this should be called right before line 186, just to be sure that we don't override c.started or c.metrics with this function call.